### PR TITLE
chore(build): move to scikitbuild-core

### PR DESCRIPTION
**Pull request by mayeut**
Depending on wether or not we want to still support "python<3.7"

close #312

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -59,6 +59,7 @@ wheel.install-dir = "cmake/data"
 
 [tool.setuptools_scm]
 write_to = "src/cmake/_version.py"
+write_to_template = "version = '{version}'"
```
Comment (jcfr) [inline pyproject.toml:62]: Are these fixes specific to python 3.5 and 2.7 ?

**Comment by jcfr** [comment]
> Depending on whether or not we want to still support "python<3.7"

Testing that the wheels works on Python 2.7 and 3.5 makes sense. Thanks for working on this :pray:

File: pyproject.toml (hunk 1 above)
Comment (mayeut) [reply pyproject.toml:62]: yes. I don't know if it's required for python 3.6 or not.

File: tests/test_cmake.py [hunk 2]
```diff
@@ -61,7 +61,9 @@ def _get_scripts():
     scripts = []
     for file in dist.files:
         if os.path.abspath(str(file.locate().parent)) in scripts_paths:
-            scripts.append(file.locate().resolve(strict=True))
+            script = file.locate().resolve()
+            assert script.exists()
+            scripts.append(script)
```
Comment (jcfr) [inline tests/test_cmake.py:66]:
```suggestion
            if sys.version_info < (3, 6):
                # pre-3.6 behavior is strict
                scripts.append(file.locate().resolve())
            else:
                scripts.append(file.locate().resolve(strict=True))
```

File: src/cmake/__init__.py [hunk 3]
```diff
@@ -12,9 +18,22 @@ def __dir__():
...
+    for script in distribution("cmake").files:
+        if str(script).startswith("cmake/data/bin/cmake"):
+            if sys.version_info < (3, 6):
+                # pre-3.6 behavior is strict
+                resolved_script = script.locate().resolve()
+            else:
+                resolved_script = script.locate().resolve(strict=True)
+            cmake_executable_path = resolved_script.parents[1]
+            break
+    CMAKE_DATA = cmake_executable_path if cmake_executable_path else None
```
Comment (jcfr) [inline src/cmake/__init__.py:32]: This adds support for editable install adding a generic way of recovering the `cmake/data` directory.

When support for Python < 3.7 will be dropped, the overall logic could be greatly simplified by directly recovering the path to `ctest`, `cmake` and `cpack` using the distribution API.

@henryiii Did I miss anything ?

**Comment by jcfr** [comment]
@mayeut @henryiii  I suggest we move forward with the integration using `Squash & Merge` using the following commit message:

```
chore(build): move to scikitbuild-core (#455)

* Maintain support for testing wheels for Python 2.7 & 3.5 (on Windows)
  by adding the `test_old_pythons` GitHub workflow job.

* Drop support for building sdist on Python 2.7 and 3.6

* Add support for editable install with Python >= 3.4 using the default "redirect" mode.

* Add support for building and testing Python 3.x wheels using Python 3.12.

* Update GitHub workflow macOS runner from `macos-11` to `macos-14`.

Co-authored-by: Henry Schreiner <HenrySchreinerIII@gmail.com>
Co-authored-by: Jean-Christophe Fillion-Robin <jchris.fillionr@kitware.com>
```

File: pyproject.toml (hunk 1 above)
Comment (henryiii) [reply pyproject.toml:62]:
```suggestion
# Setuptools-scm includes type annotations in the default template
write_to_template = "version = '{version}'"
```

**Comment by henryiii** [comment]
Decisions was to have a followup (before release) requiring 3.7+, but we could revert that if someone really needs a Python 2 or 3.6 package of the old CMake.

**Pull request merged**